### PR TITLE
Add NodeJs sdk

### PR DESCRIPTION
**Pull request by tumarsal**
Add NodeJs sdk

### CODE REVIEW

File: docs-src/content/using-cubic-sdk/recognition-context.md [hunk 1]
```diff
@@ -551,4 +551,80 @@ class CubicExample {
...
+    if (err){
+        console.error(err)
+        return
+    }
+    let models = response.getModelsList()
+    for(let model of models){
+        console.log(`${model.getId()} ${model.getName()} ${model.getAttributes()}`)
+    }
+    if (models.length>0){
+        let firstModel = models[0]
```
Comment (jhollowayj) [inline docs-src/content/using-cubic-sdk/recognition-context.md:572]:
```suggestion
        let firstModel = models[0] // Airport Names model
```

File: docs-src/content/using-cubic-sdk/recognition-context.md [hunk 2]
```diff
@@ -551,4 +551,80 @@ class CubicExample {
...
+                    if (alternatives.length>0){
+                        let words = alternatives[0].getWordsList()
+                        for (let word of words){
+                            console.log(word.getWord())
+                        }
+                    }
+                }
+            })
+
+            let config = new RecognitionConfig()
```
Comment (jhollowayj) [inline docs-src/content/using-cubic-sdk/recognition-context.md:607]:
```suggestion
            // Build and send the streaming recognize request configuration
            let config = new RecognitionConfig()
```

**Review by jhollowayj — COMMENTED** [review]
I would like to see a few more comments in the example code.
Example code should be linted and formatted (probably copy and paste it into a local file, format it, and then copy it back into the docs).

Also, the [npm package page](https://www.npmjs.com/package/@cobaltspeech/sdk-cubic) says we don't have a readme.  That might be nice to add here.

Otherwise things look like they are headed in the right direction.

File: grpc/node-cubic/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "@cobaltspeech/sdk-cubic",
+  "version": "1.0.1",
+  "description": "",
```
Comment (jhollowayj) [inline grpc/node-cubic/package.json:4]: Might be nice to add a description here.

File: grpc/node-cubic/package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "dist"
+  ],
+  "scripts": {
+    "build": "tsc --module commonjs && rollup -c rollup.config.ts",
+    "start": "rollup -c rollup.config.ts -w",
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "gen": "protoc -I . --plugin=./node_modules/.bin/protoc-gen-ts_proto --ts_proto_out=./src/grpc ../cubic.proto"
+  },
+  "keywords": [],
+  "author": "Cobalt Speech And Language, Inc.",
```
Comment (jhollowayj) [inline grpc/node-cubic/package.json:18]:
```suggestion
  "author": "Cobalt Speech & Language, Inc.",
```

File: grpc/node-cubic/package.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  "license": "Apache",
```
Comment (jhollowayj) [inline grpc/node-cubic/package.json:19]: Our top level license says Apache v2.0.  Does that matter here?  Would NodeJS treat this as Apache v1.0 instead of 2.0?

File: grpc/node-cubic/rollup.config.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import resolve from 'rollup-plugin-node-resolve'
```
[the remark this answers is not in the document]
Comment (tumarsal) [reply grpc/node-cubic/rollup.config.ts:1]: It is alternative of webpack.config.js (but it support all lib types) (es5, es6, commonjs)

File: docs-src/content/using-cubic-sdk/recognition-context.md [hunk 7]
```diff
@@ -551,4 +551,80 @@ class CubicExample {
 }
 {{< /tab >}}
 
+{{< tab "NodeJS" "js" >}}
+import {RecognitionAudio,
```
[the remark this answers is not in the document]
Comment (tumarsal) [reply docs-src/content/using-cubic-sdk/recognition-context.md:555]: I write it in vscode and there is auto linter.

File: docs-src/content/using-cubic-sdk/connecting.md [hunk 8]
```diff
@@ -176,7 +181,11 @@ CubicGrpc.CubicStub mCubicService = CubicGrpc.newStub(mCubicChannel);
 {{< tab "Swift/iOS" "swift" >}}
 let client = Client(host: serverAddress, port: serverPort, useTLS: false)
 {{< /tab >}}
-
+{{< tab "NodeJS" "js" >}}
+    import {grpc} from "@improbable-eng/grpc-web"
+    const myTransport = grpc.CrossBrowserHttpTransport({ withCredentials: true });
```
Comment (jhollowayj) [inline docs-src/content/using-cubic-sdk/connecting.md:186]:
```suggestion
    import {grpc} from "@improbable-eng/grpc-web"
    
    const myTransport = grpc.CrossBrowserHttpTransport({ withCredentials: true });
```

File: docs-src/content/using-cubic-sdk/recognition-context.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (jhollowayj) [reply docs-src/content/using-cubic-sdk/recognition-context.md:555]: Can you also format this code?  Indentation doesn't look to be consistent.

File: grpc/go-cubic/cubicpb/gw/go.mod [hunk 9]
```diff
@@ -4,9 +4,19 @@ go 1.12
 
 require (
 	github.com/cobaltspeech/sdk-cubic/grpc/go-cubic v1.6.3
+	github.com/desertbit/timer v0.0.0-20180107155436-c41aec40b27f // indirect
```
Comment (jhollowayj) [inline grpc/go-cubic/cubicpb/gw/go.mod:7]: Can we move all of the Go changes into a different PR?  Or was this needed to support the new improbable-eng/grpc-web stuff?  

If it's needed, would it be better to put that into a separate PR, merge it, and then add node support in a second PR?

**Review by jhollowayj — COMMENTED** [review]
Our current web demo for cubic already does javascript stuff.  Would you please tell me what the difference is between that and what we hope to accomplish here?  Are there extra features that this will support?  Should the web demo be using this stuff instead of whatever it was using?

I don't understand the need for doing all of this node stuff, but that doesn't mean it's not useful.

File: grpc/go-cubic/cubicpb/gw/go.mod (hunk 9 above)
Comment (tumarsal) [reply grpc/go-cubic/cubicpb/gw/go.mod]: https://github.com/cobaltspeech/sdk-cubic/pull/111

**Comment by happyalu** [comment]
Closing this.
